packages/gestalt/src/Button.js

```diff
@@ -143,6 +155,85 @@ const ButtonWithForwardRef: React$AbstractComponent<
   );
 
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        hoverStyle="none"
+        href={href}
+        ref={innerRef}
+        rel={rel}
+        target={target}
+        inline
+        rounding="pill"
+      >
+        <div
+          style={{ alignItems: 'center', display: 'flex' }}
+          className={classes}
+        >
+          {buttonText}
+        </div>
+      </Link>
+    );
+  }
+
+  const iconEndComponent = (iconEnd: $Keys<typeof icons>): Node => {
+    const SIZE_NAME_TO_PIXEL = {
+      sm: 10,
+      md: 12,
+      lg: 12,
+    };
+
+    return (
+      <Box alignItems="center" display="flex">
+        {buttonText}
+        <Box display="inlineBlock" flex="none" marginStart={2}>
+          <Icon
+            accessibilityLabel=""
+            color={textColor}
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
+  if (props.type === 'submit') {
+    const { iconEnd, name, type } = props;
+
+    return (
+      <button
+        aria-label={accessibilityLabel}
+        className={classes}
+        disabled={disabled}
+        name={name}
+        onBlur={handleBlur}
+        onClick={event => onClick && onClick({ event })}
+        onMouseDown={handleMouseDown}
+        onMouseUp={handleMouseUp}
+        onTouchCancel={handleTouchCancel}
+        onTouchEnd={handleTouchEnd}
+        onTouchMove={handleTouchMove}
+        onTouchStart={handleTouchStart}
+        ref={innerRef}
+        type={type}
+        {...(compressStyle ? { style: compressStyle } : {})}
+      >
+        {iconEnd ? iconEndComponent(iconEnd) : buttonText}
```

Comment:
@AlbertCarreras does this mean we can not have `iconEnd` on `Link`? I think that's a pretty big use case (e.g. external links that look like a button)

Reply:
Agree. While coding it got in the mindset that icons were not needed for links. Again, WIP, finishings were not done.

Reply:
DONE